File: snippets/python_snippets/connecting_lsp.py
import logging
from breez_sdk import BlockingBreezServices

sdk_services = BlockingBreezServices

def get_lsp_info():
    try: 
        # ANCHOR: get-lsp-info
        lsp_id = sdk_services.lsp_id()
        lsp_info = sdk_services.lsp_info()
        # ANCHOR_END: get-lsp-info
        return lsp_info
    except Exception as error:
        logging.log(error)
        raise

def connect_lsp(lsp_id):
    try:
        # ANCHOR: connect-lsp
        sdk_services.connect_lsp(lsp_id)
        # ANCHOR_END: connect-lsp
    except Exception as error:
        logging.log(error)
        raise

File: snippets/kotlin_mpp_lib/shared/src/commonMain/kotlin/com/example/kotlinmpplib/ConnectingLsp.kt
package com.example.kotlinmpplib

import breez_sdk.*
class ConnectingLsp {
    fun getLspInfo(sdk: BlockingBreezServices) {
        // ANCHOR: get-lsp-info
        try {
            val lspId = sdk.lspId()
            if (lspId != null) {
                val lspInfo = sdk.lspInfo()
            } else {
                // Handle no lsp id scenario
            }
        } catch (e: Exception) {
            // Handle error
        }
        // ANCHOR_END: get-lsp-info
    }

    fun listLsps(sdk: BlockingBreezServices) {
        // ANCHOR: list-lsps
        try {
            val availableLsps = sdk.listLsps()
        } catch (e: Exception) {
            // Handle error
        }
        // ANCHOR_END: list-lsps
    }
    fun connectLsp(sdk: BlockingBreezServices, lspId: String) {
        // ANCHOR: connect-lsp
        try {
            sdk.connectLsp(lspId)
        } catch (e: Exception) {
            // Handle error
        }
        // ANCHOR_END: connect-lsp
    }
}